bin/update-debian-dependencies.sh

```diff
@@ -0,0 +1,143 @@
+    
+    echo "$cosign_version"
+}
+
+# Function to update dependency in ubuntu-deps.conf
+update_dependency() {
+    local name="$1"
+    local new_release="$2"
+    local new_commit="$3"
+    local temp_file=$(mktemp)
+    
+    sed \
+        -e "s/${name}_RELEASE=\"[^\"]*\"/${name}_RELEASE=\"$new_release\"/" \
+        -e "s/${name}_COMMIT=\"[^\"]*\"/${name}_COMMIT=\"$new_commit\"/" \
+        "$UBUNTU_DEPS" > "$temp_file"
+    
+    # Use deps file permissions
+    chmod --reference="$UBUNTU_DEPS" "$temp_file"
+    mv "$temp_file" "$UBUNTU_DEPS"
+}
+
+echo "Updating dependencies in ubuntu-deps.conf..."
+
+# Update finch-daemon
+echo "Updating finch-daemon..."
+FINCHD_LATEST=$(get_latest_tag "runfinch/finch-daemon")
+FINCHD_COMMIT=$(get_commit_for_tag "runfinch/finch-daemon" "$FINCHD_LATEST")
+update_dependency "FINCHD" "$FINCHD_LATEST" "$FINCHD_COMMIT"
+
+# Update nerdctl
+echo "Updating nerdctl..."
+NERDCTL_LATEST=$(get_latest_tag "containerd/nerdctl")
+NERDCTL_COMMIT=$(get_commit_for_tag "containerd/nerdctl" "$NERDCTL_LATEST")
+update_dependency "NERDCTL" "$NERDCTL_LATEST" "$NERDCTL_COMMIT"
+
+# Get BuildKit version from nerdctl Dockerfile
+echo "Getting BuildKit version from nerdctl Dockerfile..."
+BUILDKIT_VERSION=$(get_buildkit_version "$NERDCTL_LATEST")
+
+# Get Cosign version from nerdctl Dockerfile
+echo "Getting Cosign version from nerdctl Dockerfile..."
+COSIGN_VERSION=$(get_cosign_version "$NERDCTL_LATEST")
+
+# Update buildkit with version from nerdctl
+echo "Updating buildkit to version $BUILDKIT_VERSION..."
+BUILDKIT_COMMIT=$(get_commit_for_tag "moby/buildkit" "$BUILDKIT_VERSION")
+update_dependency "BUILDKIT" "$BUILDKIT_VERSION" "$BUILDKIT_COMMIT"
+
+# Update soci-snapshotter
+echo "Updating soci-snapshotter..."
+SOCI_LATEST=$(get_latest_tag "awslabs/soci-snapshotter")
+SOCI_COMMIT=$(get_commit_for_tag "awslabs/soci-snapshotter" "$SOCI_LATEST")
+update_dependency "SOCI" "$SOCI_LATEST" "$SOCI_COMMIT"
+
+# Update CNI plugins
+echo "Updating CNI plugins..."
+CNI_LATEST=$(get_cni_plugin_version "$NERDCTL_LATEST")
+CNI_COMMIT=$(get_commit_for_tag "containernetworking/plugins" "$CNI_LATEST")
+update_dependency "CNI" "$CNI_LATEST" "$CNI_COMMIT"
+
+# Update cosign with version from nerdctl
+echo "Updating cosign to version $COSIGN_VERSION..."
+if [ -n "$COSIGN_VERSION" ]; then
+    COSIGN_COMMIT=$(get_commit_for_tag "sigstore/cosign" "$COSIGN_VERSION")
+    update_dependency "COSIGN" "$COSIGN_VERSION" "$COSIGN_COMMIT"
+else
+    echo "WARNING: Failed to extract cosign version from nerdctl Dockerfile"
+fi
+
+echo "Dependencies updated successfully!"
+echo "Updated versions:"
+echo "  finch-daemon: $FINCHD_LATEST"
+echo "  nerdctl: $NERDCTL_LATEST"
+echo "  buildkit: $BUILDKIT_VERSION"
+echo "  soci-snapshotter: $SOCI_LATEST"
+echo "  CNI plugins: $CNI_LATEST"
+echo "  cosign: $COSIGN_VERSION"
```

Comment:
nit: I don't like that this prints every time, since it could give the impression that we're updating something we aren't, but I'm fine keeping it since we need to check the latest version to update the versions, and at that point it's probably doing more work doing version checks.